add rocksdb as transactional engine (#1190)
github/gh-ost: doc/command-line-flags.md

`--ssl-key=/path/to/ssl-key.key`: SSL private key file (in PEM format).

### storage-engine

> default is `innodb`, and `rocksdb`as an optional. InnoDB and RocksDB are both transactional engines, supporting both shared and exclusive row locks.

Reviewer: ```suggestion
Default is `innodb`, and `rocksdb` support is currently experimental. InnoDB and RocksDB are both transactional engines, supporting both shared and exclusive row locks.
```

@wangzihuacool what do you think about adding a note here about RocksDB support being experimental?

Author: good suggestion

Author: good suggestion

`--ssl-key=/path/to/ssl-key.key`: SSL private key file (in PEM format).

> ### storage-engine

Reviewer 2: @wangzihuacool / @dm-2 I think we should explain more about the limitations in testing of `gh-ost` on RocksDB and mark this support as "experimental"

Lastly, explaining differences in behaviour and what operations are unsupported is useful here too. From a glance I see these types of tests disabled:
1. Foreign key
2. Generated columns
3. Spatial
4. Geometric

Am I correct to assume RocksDB does not support the functionality for these tests @wangzihuacool? Let's make this as clear as possible to the reader 🙇

Author: Yes, RocksDB does not support these features, especially Gap lock. I'll add description to make it clear to the reader.
